link/src/main/java/com/stripe/android/link/ui/signup/SignUpViewModel.kt

```diff
@@ -0,0 +1,195 @@
+        initialValue = args.configuration.customerInfo.email
+    )
+    val phoneNumberController = PhoneNumberController.createPhoneNumberController(
+        initialValue = args.configuration.customerInfo.phone.orEmpty(),
+        initiallySelectedCountryCode = args.configuration.customerInfo.billingCountryCode
+    )
+    val nameController = NameConfig.createController(
+        initialValue = args.configuration.customerInfo.name
+    )
+    private val _state = MutableStateFlow(
+        value = SignUpScreenState(
+            signUpEnabled = false
+        )
+    )
+
+    val state: StateFlow<SignUpScreenState> = _state
+
+    private val requiresNameCollection: Boolean
+        get() {
+            val countryCode = when (val stripeIntent = args.configuration.stripeIntent) {
+                is PaymentIntent -> stripeIntent.countryCode
+                is SetupIntent -> stripeIntent.countryCode
+            }
+            return countryCode != CountryCode.US.value
+        }
+
+    init {
+        viewModelScope.launch {
+            signUpEnabledListener()
+        }
+        viewModelScope.launch {
+            emailListener()
+        }
+        linkEventsReporter.onSignupFlowPresented()
+    }
+
+    private suspend fun signUpEnabledListener() {
+        combine(
+            flow = nameController.fieldState.map {
+                if (requiresNameCollection) {
+                    it.isValid()
+                } else {
+                    true
+                }
+            },
+            flow2 = emailController.fieldState.map { it.isValid() },
+            flow3 = phoneNumberController.isComplete
+        ) { nameComplete, emailComplete, phoneComplete ->
+            nameComplete && emailComplete && phoneComplete
+        }.collectLatest { formValid ->
+            updateState { it.copy(signUpEnabled = formValid) }
+        }
+    }
+
+    private suspend fun emailListener() {
+        emailController.formFieldValue.mapLatest { entry ->
+            entry.takeIf { it.isComplete }?.value
+        }.collectLatest { email ->
+            delay(LOOKUP_DEBOUNCE)
+            if (email != null) {
+                updateSignUpState(SignUpState.VerifyingEmail)
+                lookupConsumerEmail(email)
+            } else {
+                updateSignUpState(SignUpState.InputtingPrimaryField)
+            }
+        }
+    }
+
+    fun onSignUpClick() {
+        clearError()
+        viewModelScope.launch {
+            linkAccountManager.signUp(
+                email = emailController.fieldValue.value,
+                phone = phoneNumberController.fieldValue.value,
+                country = phoneNumberController.getCountryCode(),
+                name = nameController.fieldValue.value,
+                consentAction = SignUpConsentAction.Implied
+            ).fold(
+                onSuccess = {
+                    onAccountFetched(it)
+                    linkEventsReporter.onSignupCompleted()
```

Comment:
(for a follow up PR) should the analytics be handled as part of link account manager instead of here? I'd expect the analytics to be the same wherever linkAccountManager.signUp is called from